resolver/resolver.go

```diff
@@ -38,7 +39,8 @@ func New(version string, config records.Config) *Resolver {
 		version: version,
 		config:  config,
 		rs:      &records.RecordGenerator{},
-		rng:     rand.New(rand.NewSource(time.Now().UnixNano())),
+		// See: https://github.com/golang/go/issues/3611
+		rng:     rand.New(&lockedSource{src: rand.NewSource(time.Now().UnixNano())}),
```

Comment:
Are there multiple seed calls? Why is that a race? Feel very strange to lock a rand source. This should be a singleton, injected by the main function of the binary or by tests.